libs/media-player/src/lib/timeline.tsx

```diff
@@ -0,0 +1,501 @@
+    overflowX: 'scroll',
+    scrollBehavior: 'smooth',
+    position: 'relative',
+    boxSizing: 'border-box',
+});
+
+const StyledTrackLabel = styled(Box)({
+    height: `${EDITOR_SOUND_BAR_HEIGHT_IN_PIXELS}px`,
+    padding: '2px',
+    wordWrap: 'break-word',
+    overflow: 'hidden',
+    boxSizing: 'border-box',
+    cursor: 'pointer',
+});
+
+const StyledScrolledTrack = styled(Box)({
+    minWidth: '100%',
+    position: 'relative',
+    display: 'flex',
+    height: `${EDITOR_SOUND_BAR_HEIGHT_IN_PIXELS}px`,
+    border: '1px solid #666',
+    borderRadius: '7px',
+    backgroundBlendMode: 'saturation',
+    backgroundSize: `auto ${EDITOR_SOUND_BAR_HEIGHT_IN_PIXELS + 5}px`,
+    boxSizing: 'border-box',
+});
+
+const StyledTimelineRulerBox = styled('div')({
+    marginBottom: '3px',
+    boxSizing: 'border-box',
+    position: 'relative',
+});
+
+const EditorPlayhead = styled('div')({
+    top: '0px',
+    width: '1px',
+    backgroundColor: 'red',
+    position: 'absolute',
+    // animationName: `${movePlayhead}`,
+    // animationDuration: '2s',
+    // animationTimingFunction: 'ease-in',
+    // animationIterationCount: 'infinite',
+    boxSizing: 'border-box',
+    zIndex: 1500,
+});
+
+const EditorInPoint = styled('div')({
+    top: '0px',
+    width: `${EDITOR_MARKER_PIXEL_WIDTH}px`,
+    backgroundImage: 'linear-gradient(to right, #75ecff, rgba(255,0,0,0))',
+    borderLeft: '1px solid #007082',
+    borderTopLeftRadius: '7px',
+    borderBottomLeftRadius: '7px',
+    position: 'absolute',
+    boxSizing: 'border-box',
+    visibility: 'hidden',
+    zIndex: 1500,
+});
+
+const EditorOutPoint = styled('div')({
+    top: '0px',
+    width: `${EDITOR_MARKER_PIXEL_WIDTH}px`,
+    backgroundImage: 'linear-gradient(to left, #75ecff, rgba(255,0,0,0))',
+    borderRight: '1px solid #007082',
+    borderTopRightRadius: '7px',
+    borderBottomRightRadius: '7px',
+    position: 'absolute',
+    boxSizing: 'border-box',
+    visibility: 'hidden',
+    zIndex: 1500,
+});
+
+export type TimeRangeSeconds = {
+    inPointSeconds: number;
+    outPointSeconds: number;
+};
+
+export interface TimeRangeClip {
+    timeRangeSeconds: TimeRangeSeconds;
+    label: React.ReactNode;
+}
+
+// Also in interactive-annotator.tsx
+export enum TimelineTrackName {
+    annotations = 'annotations',
+    transcriptions = 'transcriptions',
+}
+
+export type TimelineTrack = {
+    name: TimelineTrackName;
+    trackLabel: string;
+    timelineTrack: TimeRangeClip[];
+};
+
+interface TimelineProps {
+    durationSeconds: number;
+    timelineTracks: TimelineTrack[];
+    name: string;
+    audioRef: RefObject<HTMLAudioElement>;
+    isPlaying: boolean;
+    seekInMedia: (newTime: number) => void;
+    inPointSeconds: number | null;
+    outPointSeconds: number | null;
+    mediaCurrentTimeFromContext: number;
+    timelineTrackName: TimelineTrackName;
+    setTimelineTrackName: (trackName: TimelineTrackName) => void;
+}
+
+export const Timeline = ({
```

Comment:
There's way too much going on in this component. The cognitive overload makes it very hard to parse. Let's see if we can use some function decomposition on this one.